packages/dart_frog_cli/lib/src/prod_server_builder/prod_server_builder.dart

```diff
@@ -0,0 +1,81 @@
+import 'dart:io' as io;
+
+import 'package:dart_frog_cli/src/runtime_compatibility.dart';
+import 'package:mason/mason.dart';
+import 'package:meta/meta.dart';
+
+/// Typedef for [ProdServerBuilder.new].
+typedef ProdServerBuilderBuilder = ProdServerBuilder Function({
+  required Logger logger,
+  required String dartVersion,
+  required io.Directory workingDirectory,
+  required MasonGenerator prodServerBundleGenerator,
+});
+
+/// {@template prod_server_builder}
+/// A class that builds the production server runtime code.
+///
+/// This class is responsible for:
+///  - Ensuring that the current version of `package:dart_frog` is compatible
+///    with the generated code.
+///  - Generating the production server runtime code.
+///
+/// {@endtemplate}
+class ProdServerBuilder {
```

Comment:
Maybe we can rename it to ProdServerGenerator to avoid the double builder?

Reply:
It seems aligned with the build command to build a production server. The generator messes with the mason terminology (it is even more confusing because then the generator would receive a mason generator in its constructor).

Reply:
I see, the name clash is a valid concern. I'm happy with both the current BuilderBuilder or the new BuilderConstructor.